src/Query/Mysql/Acos.php

```diff
@@ -20,11 +21,13 @@ public function getSql(SqlWalker $sqlWalker): string
 
     public function parse(Parser $parser): void
     {
-        $parser->match(Lexer::T_IDENTIFIER);
-        $parser->match(Lexer::T_OPEN_PARENTHESIS);
+        $orm3 = !defined('\Doctrine\ORM\Query\Lexer::T_IDENTIFIER');
+        
+        $parser->match($orm3 ? TokenType::T_IDENTIFIER : Lexer::T_IDENTIFIER);
```

Comment:
@greg0ire looking at all this conditional code, would it make sense to backport `TokenType` to ORM 2.x?

Reply:
I think it might, but `TokenType` is an enum, so there would be a type change from class to enum between 2 and 3, which is probably why I didn't introduce it in 2.x.

Reply:
@greg0ire Can `TokenType` in ORM2 be a class that becomes an enum in ORM3? The notation is similar, so nothing changes in the direction of using the library.

Reply:
Maybe yes. We could add a private constructor or a constructor that always throws to prevent attempts at instantiating the class.

Reply:
@greg0ire @derrabus This solution really makes ORM2 and ORM3 support more user friendly.

The new class:
https://github.com/doctrine/orm/commit/bd0a198e12f0fbb3784f7ec70c1ca4b58ab90b36
And the usage of the class:
https://github.com/beberlei/DoctrineExtensions/blob/a5fba21f0c502591dc682370b546d8a679871eb8/src/Query/Mysql/Acos.php

Reply:
Please send a PR to the ORM with that new class.

Reply:
@derrabus Should I follow the instructions in the "How to choose the right branch" issue and do it in 2.19.x, or can I go down in version? It really doesn't affect anything if we include the class in earlier versions and it helps for example this library to support as many versions as possible.